README.md

```diff
@@ -40,141 +40,141 @@ npm run lint-fix # format source files
-| [#240][#240] | [SHOULD declare enum values using UPPER_SNAKE_CASE format][#240] | :heavy_check_mark: |
-| [#241][#241] | [MAY expose compound keys as resource identifiers][#241] | - |
+| id | title |supported | prio (A, B, C)|
+| --- | --- | --- | --- |
+| [#100][#100] | [MUST follow API first principle][#100] | - | B |
+| [#101][#101] | [MUST provide API specification using OpenAPI][#101] | - | A |
+| [#102][#102] | [SHOULD provide API user manual][#102] | - | C |
+| [#103][#103] | [MUST write APIs using U.S. English][#103] | - | A |
+| [#104][#104] | [MUST secure endpoints with OAuth 2.0][#104] | - | ? |
+| [#105][#105] | [MUST define and assign permissions (scopes)][#105] | - | B |
+| [#106][#106] | [MUST not break backward compatibility][#106] | - | B |
+| [#107][#107] | [SHOULD prefer compatible extensions][#107] | - | ? |
+| [#108][#108] | [MUST prepare clients accept compatible API extensions][#108] | - | C |
+| [#109][#109] | [SHOULD design APIs conservatively][#109] | - | ? |
+| [#110][#110] | [MUST always return JSON objects as top-level data structures][#110] | :heavy_check_mark: | A |
+| [#111][#111] | [MUST treat Open API specification as open for extension by default][#111] | - | - |
+| [#112][#112] | [SHOULD used open-ended list of values (`x-extensible-enum`) for enumerations][#112] | :heavy_check_mark: | - |
+| [#113][#113] | [SHOULD avoid versioning][#113] | - | B |
+| [#114][#114] | [MUST use media type versioning][#114] | - | - |
+| [#115][#115] | [MUST not use URI versioning][#115] | - | A |
+| [#116][#116] | [MUST use semantic versioning][#116] | :heavy_check_mark: | B |
+| [#118][#118] | [MUST property names must be ASCII snake_case (and never camelCase): `^[a-z_][a-z_0-9]*$`][#118] | :heavy_check_mark: | A |
+| [#120][#120] | [SHOULD pluralize array names][#120] | - | B |
+| [#122][#122] | [MUST not use `null` for boolean properties][#122] | - | - |
+| [#123][#123] | [MUST use same semantics for `null` and absent properties][#123] | - | - |
+| [#124][#124] | [SHOULD not use `null` for empty arrays][#124] | - | - |
+| [#125][#125] | [SHOULD represent enumerations as strings][#125] | - | - |
+| [#126][#126] | [SHOULD define dates properties compliant with RFC 3339][#126] | - | - |
+| [#127][#127] | [SHOULD define time durations and intervals properties conform to ISO 8601][#127] | - | - |
+| [#128][#128] | [SHOULD use standards for country, language and currency codes][#128] | - | - |
+| [#129][#129] | [MUST use lowercase separate words with hyphens for path segments][#129] | - | - |
+| [#130][#130] | [MUST use snake_case (never camelCase) for query parameters][#130] | - | B |
+| [#132][#132] | [SHOULD prefer hyphenated-pascal-case for HTTP header fields][#132] | - | - |
+| [#133][#133] | [MAY use standardized headers][#133] | - | - |
+| [#134][#134] | [MUST pluralize resource names][#134] | - | A |
+| [#135][#135] | [SHOULD not use /api as base path][#135] | - | ? |
```

Comment:
```suggestion
| [#135][#135] | [SHOULD not use /api as base path][#135] | - | A |
```
This is easy to implement

Reply:
But do we not want this? Our api uses always /api/v1/users/1 .

Reply:
It's a SHOULD. So only a warning.
Your example also violates "MUST not use URI versioning [115]". Using "/api" in the URL only makes sense for e.g. backend apis where the frontend is served under "/". Those backend apis are hopefully only used by the frontend and are therefore not a shared service. Standalone services can easily follow the zalando guide. I think thats also what they refer to.

Reply:
Intresting point. In our case we add the /api with openshift deployment. Currently there is only one consumer and this is our client. In future case I expect an service API which handles the url for me so I would expect an url like /sales/partner/users/1, something like a pattern /{domain}/{boundedContext}/{ressource}. What do you think about the version? Where is the version present? Do you need it in our http request in the header or as query parameters?

Reply:
Zalando puts the domain etc. into the hostname (https://opensource.zalando.com/restful-api-guidelines/#224). If we locate all services behind some sort of gateway with a single hostname we have to deviate from that rule and move the domain into the path, true.

Regarding the versioning:
[SHOULD avoid versioning [113]](https://opensource.zalando.com/restful-api-guidelines/#113)
```
If changing an API can’t be done in a compatible way, then proceed in one of these three ways:
 - create a new resource (variant) in addition to the old resource variant
 - create a new service endpoint — i.e. a new application with a new API (with a new domain name)
 - create a new API version supported in parallel with the old API by the same microservice
```
The third option is then done via headers: [MUST use media type versioning [114]](https://opensource.zalando.com/restful-api-guidelines/#114)